pkg/layer/group_test.go

```diff
@@ -135,3 +139,15 @@ func TestGroup_MoveLayerFrom(t *testing.T) {
 	group.MoveLayerFrom(l1, 1, &group)
 	assert.Equal(t, l1, group.Layers().Layers()[1])
 }
+
+func TestGroup_Tags(t *testing.T) {
+	tt := id.NewTagID()
+	err := group.AttachTag(tt)
+	assert.NoError(t, err)
+	tl := tags
+	tl = append(tl, tt)
+	assert.Equal(t, tl, group.Tags().Tags())
+	err = group.DetachTag(tt)
+	assert.NoError(t, err)
+	assert.Equal(t, tags, group.Tags().Tags())
+}
```

Comment:
I found that this test file is not very good, because multiple functions are referencing and mutating a single variable. In general, in many cases, code that mutates global variables (or their equivalents, in this case variables defined with var) is an anti-pattern to avoid. It is true that Go's unit tests are executed sequentially within the same package and no problem to run the tests, but the independence of functions is compromised and the implementation becomes difficult to understand.

But this time, let's leave it at that. We'll have a chance to refactor it later.